### PR TITLE
Integrate Sections tab and navigation buttons

### LINKED ISSUE #684: Profile section button
### Overview
Create the profile section button and render it a list of the on the sidebar.

For this, you will have to work off the `feat/generated-profile` branch.

Screenshot 2024-04-23 at 6 25 01 PM

Screenshot 2024-04-23 at 6 24 45 PM

### Action Items

- [x] Create a profile section button component
- [x] The button should work for the 4 states: unselected, selected, disabled, partial.
- [x] Have the sidebar render one for each section

Also wondering if we should just leave these as links or use a Tab component from MUI:
https://mui.com/material-ui/react-tabs/#nav-tabs

### Resources/Instructions
Figma frame:
https://www.figma.com/file/BNWqZk8SHKbtN1nw8BB7VM/HUU-Everything-Figma-Nov-2022?type=design&node-id=11672-90757&mode=design&t=IekRfD8VxWN7bKYK-4

### PR DESCRIPTION
**Pull request by JpadillaCoding**
Closes #684 

### What changes did you make?
  - Made the Sections tab ``sections.tsx`` ``\intakeProfile\index.tsx`` ``inProgressIcon.tsx`` ``lockedIcon.tsx``
  - Added styling based on the status of the section task ``useStatusStyling.tsx``
  - Added styling to nav buttons and made them responsive
  - Nav buttons work for moving throughout the form
  - Cleaned up files by removing application Context wrapper along with old sections component ``\guestApplicationForm`` ``GuestApplicationSchema.ts`` 
### Rationale behind the changes?
  - Styling for the status is set with the intent of when a user's profile is pulled and compared to ``fieldGroups``, they'll have incorporated data on the status of each task which will then render the type of styling.
  - Since we are now using ``IntakeProfile`` to control the profile form, the previous context approach and related files were removed.
  - Basic functionality for the nav buttons to allow navigating the form. Currently, the only option I found is obtaining the index of the current groupID and moving to the next index item on the fieldGroup object. I am open to any other approaches to this, but it is functional as is.
### Testing done for these changes
  - Changing the status on the sections tab between the 4 options of ``complete | incomplete | locked | partial``
  - Testing on mobile, desktop, and tablet view to check layout and button responsiveness.
  - Tested the buttons by going back and forth including trying to get out of bounds on the first and last item
### What did you learn or can share that is new?(optional)
- Being able to use breakpoints on MUI. This is an easy way to make an application responsive based on the screen view size. 

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)

<details>
<summary>Visuals before changes are applied</summary>

Screenshot 2024-06-02 at 3 29 21 PM

Screenshot 2024-06-02 at 3 28 51 PM

</details>

<details>
<summary>Visuals after changes are applied</summary>
  
Screenshot 2024-06-02 at 3 43 16 PM

Screenshot 2024-06-05 at 8 25 25 PM

Screenshot 2024-06-05 at 8 27 05 PM

</details>

### CODE REVIEW

File: app/src/views/IntakeProfile/index.tsx [hunk 1]
```diff
@@ -17,15 +17,27 @@ export type InitialValues = Record<string, Values>;
...
     {skip: !profileId},
   );
   const {data: responsesData} = useGetResponsesQuery({userId: '1'});
 
+  useEffect(() => {
+    if (groupId) {
+      setSelectedItem(groupId);
+    }
+  }, [groupId]);
+
```
Comment (erikguntner) [inline app/src/views/IntakeProfile/index.tsx:40]: For the selected item check, could we pass the `groupId` accessed in the params to the sidebar and do an equality check against the field group id? That way we can treat the groupId from useParams like the state variable.

**Review by erikguntner — COMMENTED** [review]
Thanks @JpadillaCoding the updates are looking great! I have a few suggestions. Let me know what you think.

File: app/src/views/IntakeProfile/index.tsx (hunk 1 above)
Comment (JpadillaCoding) [reply app/src/views/IntakeProfile/index.tsx]: Nice catch for some optimization! Changes made.

File: app/src/components/Icons/LockedIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+export const LockedIcon = () => {
```
[the remark this answers is not in the document]
Comment (JpadillaCoding) [reply app/src/components/Icons/LockedIcon.tsx]: I changed the icon to use the MUi import instead and had to include some styling to make the circle background.

**Review by erikguntner — COMMENTED** [review]
Thanks for the updates @JpadillaCoding! Merging now.

**Review by erikguntner — APPROVED** [review]

**Pull request merged**